daemon/src/auto_rollover.rs

```diff
@@ -73,68 +70,71 @@ where
             .expect("actor to be able to give address to itself");
 
         for id in cfd_ids {
-            let disconnected = match self.rollover_actors.get_disconnected(id) {
-                Ok(disconnected) => disconnected,
-                Err(_) => {
-                    tracing::debug!(order_id=%id, "Rollover already in progress");
-                    continue;
-                }
-            };
-
-            // TODO: Shall this have a try_continue?
-            let cfd = load_cfd(id, &mut conn).await?;
-
-            let (addr, fut) = rollover_taker::Actor::new(
-                (cfd, self.n_payouts),
-                self.oracle_pk,
-                self.conn_actor.clone(),
-                &self.oracle_actor,
-                self.projection_actor.clone(),
-                &this,
-                (&this, &self.conn_actor),
-            )
-            .create(None)
-            .run();
+            let cfd = try_continue!(load_cfd(id, &mut conn).await);
+
+            if let Err(e) = cfd.can_auto_rollover_taker(OffsetDateTime::now_utc()) {
```

Comment:
Because we have introduced the indirection of the `auto_rollover::Actor` sending a message to itself to spawn a `rollover_taker::Actor` we can actually directly send said message in the actor tests, circumventing this check.